src/gui/qgsattributeform.cpp

```diff
@@ -1417,6 +1483,32 @@ void QgsAttributeForm::init()
     }
   }
 
+  //create defaultValueDependencies
+  for ( QgsWidgetWrapper *ww : qgis::as_const( mWidgets ) )
+  {
+    QgsEditorWidgetWrapper *eww = qobject_cast<QgsEditorWidgetWrapper *>( ww );
+    if ( eww )
+    {
+      QgsExpression exp( eww->field().defaultValueDefinition().expression() );
+      for ( const QString &referencedColumn : exp.referencedColumns().toList() )
```

Comment:
```suggestion
      const QList<QString> referencedColumns = exp.referencedColumns().toList();
      for ( const QString &referencedColumn : referencedColumns )
```

... or the list will detach. I wonder if you could directly iterate the QSet and avoid the conversion to a list too?

Reply:
yes